Windows failure:
```
C:\Users\swift-ci\jenkins\workspace\swift-PR-windows\sourcekit-lsp\Sources\LanguageServerProtocolJSONRPC\JSONRPCConnection.swift:370:50: error: type '(RequestID) -> Optional<any ResponseType.Type>' does not conform to the 'Sendable' protocol
368 | 
369 |     // Setup callback for response type.
370 |     decoder.userInfo[.responseTypeCallbackKey] = { (id: RequestID) -> ResponseType.Type? in
    |                                                  |- error: type '(RequestID) -> Optional<any ResponseType.Type>' does not conform to the 'Sendable' protocol

    |                                                  `- note: a function type must be marked '@Sendable' to conform to 'Sendable'

371 |       guard let outstanding = self.outstandingRequests[id] else {

372 |         logger.error("Unknown request for \(id, privacy: .public)")
```

--- a/Runtimes/Core/CMakeLists.txt
+++ b/Runtimes/Core/CMakeLists.txt
@@ -83,6 +83,7 @@ include(EmitSwiftInterface)
 include(PlatformInfo)
 include(gyb)
 include(ResourceEmbedding)
+include(CatalystSupport)
 
 check_symbol_exists("asl_log" "asl.h" SwiftCore_HAS_ASL)
 check_symbol_exists("dladdr" "dlfcn.h" SwiftCore_HAS_DLADDR)

--- a/Runtimes/Core/cmake/caches/Vendors/Apple/arm64e-MacOSX.cmake
+++ b/Runtimes/Core/cmake/caches/Vendors/Apple/arm64e-MacOSX.cmake
@@ -2,6 +2,10 @@ if(NOT DEFINED CMAKE_OSX_DEPLOYMENT_TARGET)
   message(SEND_ERROR "CMAKE_OSX_DEPLOYMENT_TARGET not defined")
 endif()
 
+if(NOT DEFINED SwiftCore_TARGET_VARIANT_DEPLOYMENT_TARGET)
+  message(WARNING "SwiftCore_TARGET_VARIANT_DEPLOYMENT_TARGET not defined")
+endif()
+
 set(CMAKE_C_COMPILER_TARGET "arm64e-apple-macosx${CMAKE_OSX_DEPLOYMENT_TARGET}" CACHE STRING "")
 set(CMAKE_CXX_COMPILER_TARGET "arm64e-apple-macosx${CMAKE_OSX_DEPLOYMENT_TARGET}" CACHE STRING "")
 set(CMAKE_Swift_COMPILER_TARGET "arm64e-apple-macosx${CMAKE_OSX_DEPLOYMENT_TARGET}" CACHE STRING "")
@@ -9,8 +13,6 @@ set(CMAKE_Swift_COMPILER_TARGET "arm64e-apple-macosx${CMAKE_OSX_DEPLOYMENT_TARGE
 set(SwiftCore_ARCH_SUBDIR arm64e CACHE STRING "")
 set(SwiftCore_PLATFORM_SUBDIR macosx CACHE STRING "")
 
-list(APPEND CMAKE_C_FLAGS "-darwin-target-variant" "arm64e-apple-ios13.1-macabi")
-list(APPEND CMAKE_CXX_FLAGS "-darwin-target-variant" "arm64e-apple-ios13.1-macabi")
-list(APPEND CMAKE_Swift_FLAGS "-target-variant" "arm64e-apple-ios13.1-macabi")
+set(SwiftCore_COMPILER_VARIANT_TARGET "arm64e-apple-ios${SwiftCore_TARGET_VARIANT_DEPLOYMENT_TARGET}-macabi" CACHE STRING "")
 
 include("${CMAKE_CURRENT_LIST_DIR}/apple-common.cmake")

--- a/Runtimes/Core/cmake/caches/Vendors/Apple/x86_64-MacOSX.cmake
+++ b/Runtimes/Core/cmake/caches/Vendors/Apple/x86_64-MacOSX.cmake
@@ -2,6 +2,10 @@ if(NOT DEFINED CMAKE_OSX_DEPLOYMENT_TARGET)
   message(SEND_ERROR "CMAKE_OSX_DEPLOYMENT_TARGET not defined")
 endif()
 
+if(NOT DEFINED SwiftCore_TARGET_VARIANT_DEPLOYMENT_TARGET)
+  message(WARNING "SwiftCore_TARGET_VARIANT_DEPLOYMENT_TARGET not defined")
+endif()
+
 set(CMAKE_C_COMPILER_TARGET "x86_64-apple-macosx${CMAKE_OSX_DEPLOYMENT_TARGET}" CACHE STRING "")
 set(CMAKE_CXX_COMPILER_TARGET "x86_64-apple-macosx${CMAKE_OSX_DEPLOYMENT_TARGET}" CACHE STRING "")
 set(CMAKE_Swift_COMPILER_TARGET "x86_64-apple-macosx${CMAKE_OSX_DEPLOYMENT_TARGET}" CACHE STRING "")
@@ -9,8 +13,6 @@ set(CMAKE_Swift_COMPILER_TARGET "x86_64-apple-macosx${CMAKE_OSX_DEPLOYMENT_TARGE
 set(SwiftCore_ARCH_SUBDIR x86_64 CACHE STRING "")
 set(SwiftCore_PLATFORM_SUBDIR macosx CACHE STRING "")
 
-list(APPEND CMAKE_C_FLAGS "-darwin-target-variant" "x86_64-apple-ios13.1-macabi")
-list(APPEND CMAKE_CXX_FLAGS "-darwin-target-variant" "x86_64-apple-ios13.1-macabi")
-list(APPEND CMAKE_Swift_FLAGS "-target-variant" "x86_64-apple-ios13.1-macabi")
+set(SwiftCore_COMPILER_VARIANT_TARGET "x86_64-apple-ios${SwiftCore_TARGET_VARIANT_DEPLOYMENT_TARGET}-macabi" CACHE STRING "")
 
 include("${CMAKE_CURRENT_LIST_DIR}/apple-common.cmake")

--- a/Runtimes/Core/cmake/modules/CatalystSupport.cmake
+++ b/Runtimes/Core/cmake/modules/CatalystSupport.cmake
@@ -1,0 +1,19 @@
+# Add flags for generating the zippered target variant in the build
+
+if(SwiftCore_COMPILER_VARIANT_TARGET)
+  add_compile_options(
+    "$<$<COMPILE_LANGUAGE:C,CXX>:SHELL:-darwin-target-variant ${SwiftCore_COMPILER_VARIANT_TARGET}>"
+    "$<$<COMPILE_LANGUAGE:Swift>:SHELL:-target-variant ${SwiftCore_COMPILER_VARIANT_TARGET}>"
+
+    # TODO: Remove me once we have a driver with
+    #   https://github.com/swiftlang/swift-driver/pull/1803
+    "$<$<COMPILE_LANGUAGE:Swift>:SHELL:-Xclang-linker -darwin-target-variant -Xclang-linker ${SwiftCore_COMPILER_VARIANT_TARGET}>")
+
+  add_link_options(
+    "$<$<LINK_LANGUAGE:C,CXX>:SHELL:-darwin-target-variant ${SwiftCore_COMPILER_VARIANT_TARGET}>"
+    "$<$<LINK_LANGUAGE:Swift>:SHELL:-target-variant ${SwiftCore_COMPILER_VARIANT_TARGET}>"
+
+    # TODO: Remove me once we have a driver with
+    #   https://github.com/swiftlang/swift-driver/pull/1803
+    "$<$<LINK_LANGUAGE:Swift>:SHELL:-Xclang-linker -darwin-target-variant -Xclang-linker ${SwiftCore_COMPILER_VARIANT_TARGET}>")
+endif()

--- a/Runtimes/Core/cmake/modules/CatalystSupport.cmake
+++ b/Runtimes/Core/cmake/modules/CatalystSupport.cmake
@@ -1,5 +1,8 @@
 # Add flags for generating the zippered target variant in the build
 
+# Initialize `SwiftCore_VARIANT_MODULE_TRIPLE` if the driver is able to emit
+# modules for the target variant.
+
 if(SwiftCore_COMPILER_VARIANT_TARGET)
   add_compile_options(
     "$<$<COMPILE_LANGUAGE:C,CXX>:SHELL:-darwin-target-variant ${SwiftCore_COMPILER_VARIANT_TARGET}>"
@@ -16,4 +19,20 @@ if(SwiftCore_COMPILER_VARIANT_TARGET)
     # TODO: Remove me once we have a driver with
     #   https://github.com/swiftlang/swift-driver/pull/1803
     "$<$<LINK_LANGUAGE:Swift>:SHELL:-Xclang-linker -darwin-target-variant -Xclang-linker ${SwiftCore_COMPILER_VARIANT_TARGET}>")
+
+  # TODO: Once we are guaranteed to have a driver with the variant module path
+  #       support everywhere, we should integrate this into PlatformInfo.cmake
+  check_compiler_flag(Swift "-emit-variant-module-path ${CMAKE_CURRENT_BINARY_DIR}/CompilerID/variant.swiftmodule" HAVE_Swift_VARIANT_MODULE_PATH_FLAG)
+  if(HAVE_Swift_VARIANT_MODULE_PATH_FLAG)
+    # Get variant module triple
+    set(module_triple_command "${CMAKE_Swift_COMPILER}" -print-target-info -target ${SwiftCore_COMPILER_VARIANT_TARGET})
+    execute_process(COMMAND ${module_triple_command} OUTPUT_VARIABLE target_info_json)
+    message(CONFIGURE_LOG "Swift target variant info: ${target_info_json}")
+
+
+    string(JSON module_triple GET "${target_info_json}" "target" "moduleTriple")
+    set(SwiftCore_VARIANT_MODULE_TRIPLE "${module_triple}" CACHE STRING "Triple used for installed swift{module,interface} files for the target variant")
+    mark_as_advanced(SwiftCore_VARIANT_MODULE_TRIPLE)
+    message(CONFIGURE_LOG "Swift target variant module triple: ${module_triple}")
+  endif()
 endif()

--- a/Runtimes/Core/cmake/modules/EmitSwiftInterface.cmake
+++ b/Runtimes/Core/cmake/modules/EmitSwiftInterface.cmake
@@ -12,6 +12,16 @@ function(emit_swift_interface target)
       $<$<COMPILE_LANGUAGE:Swift>:-emit-private-module-interface-path$<SEMICOLON>${CMAKE_CURRENT_BINARY_DIR}/$<TARGET_PROPERTY:${target},Swift_MODULE_NAME>.private.swiftinterface>
       $<$<COMPILE_LANGUAGE:Swift>:-library-level$<SEMICOLON>api>
       $<$<COMPILE_LANGUAGE:Swift>:-Xfrontend$<SEMICOLON>-require-explicit-availability=ignore>)
+
+      # Emit catalyst swiftmodules and interfaces
+      if(SwiftCore_VARIANT_MODULE_TRIPLE)
+        set(variant_module_tmp_dir "${CMAKE_CURRENT_BINARY_DIR}/${target}-${SwiftCore_VARIANT_MODULE_TRIPLE}")
+        file(MAKE_DIRECTORY "${variant_module_tmp_dir}")
+        target_compile_options(${target} PRIVATE
+          "$<$<COMPILE_LANGUAGE:Swift>:SHELL:-emit-variant-module-path ${variant_module_tmp_dir}/${target}.swiftmodule>"
+          "$<$<COMPILE_LANGUAGE:Swift>:SHELL:-emit-variant-module-interface-path ${variant_module_tmp_dir}/${target}.swiftinterface>"
+          "$<$<COMPILE_LANGUAGE:Swift>:SHELL:-emit-variant-private-module-interface-path ${variant_module_tmp_dir}/${target}.private.swiftinterface>")
+      endif()
   endif()
 endfunction()
 
@@ -26,5 +36,18 @@ function(install_swift_interface target)
     install(FILES "${CMAKE_CURRENT_BINARY_DIR}/$<TARGET_PROPERTY:${target},Swift_MODULE_NAME>.private.swiftinterface"
       RENAME "${SwiftCore_MODULE_TRIPLE}.private.swiftinterface"
       DESTINATION "${SwiftCore_INSTALL_SWIFTMODULEDIR}/$<TARGET_PROPERTY:${target},Swift_MODULE_NAME>.swiftmodule")
+
+    # Install catalyst interface files
+    if(SwiftCore_VARIANT_MODULE_TRIPLE)
+      install(FILES "${CMAKE_CURRENT_BINARY_DIR}/${target}-${SwiftCore_VARIANT_MODULE_TRIPLE}/${target}.swiftmodule"
+        RENAME "${SwiftCore_VARIANT_MODULE_TRIPLE}.swiftmodule"
+        DESTINATION "${SwiftCore_INSTALL_SWIFTMODULEDIR}/$<TARGET_PROPERTY:${target},Swift_MODULE_NAME>.swiftmodule")
+      install(FILES "${CMAKE_CURRENT_BINARY_DIR}/${target}-${SwiftCore_VARIANT_MODULE_TRIPLE}/${target}.swiftinterface"
+        RENAME "${SwiftCore_VARIANT_MODULE_TRIPLE}.swiftinterface"
+        DESTINATION "${SwiftCore_INSTALL_SWIFTMODULEDIR}/$<TARGET_PROPERTY:${target},Swift_MODULE_NAME>.swiftmodule")
+      install(FILES "${CMAKE_CURRENT_BINARY_DIR}/${target}-${SwiftCore_VARIANT_MODULE_TRIPLE}/${target}.private.swiftinterface"
+        RENAME "${SwiftCore_VARIANT_MODULE_TRIPLE}.private.swiftinterface"
+        DESTINATION "${SwiftCore_INSTALL_SWIFTMODULEDIR}/$<TARGET_PROPERTY:${target},Swift_MODULE_NAME>.swiftmodule")
+    endif()
   endif()
 endfunction()

